miasm2/core/asmblock.py

```diff
@@ -166,23 +129,24 @@ def addto(self, c):
         assert isinstance(self.bto, set)
         self.bto.add(c)
 
-    def split(self, offset, l):
+    def split(self, symbol_pool, offset, loc_key):
         log_asmblock.debug('split at %x', offset)
         i = -1
         offsets = [x.offset for x in self.lines]
-        if not l.offset in offsets:
+        loc_key_offset = symbol_pool.loc_key_to_offset(loc_key)
+        if loc_key_offset not in offsets:
             log_asmblock.warning(
                 'cannot split bloc at %X ' % offset +
                 'middle instruction? default middle')
             offsets.sort()
             return None
-        new_bloc = AsmBlock(l)
+        new_bloc = AsmBlock(loc_key)
```

Comment:
Why `loc_key` arg is needed, instead of using the loc_key corresponding to `offset`?

Reply:
done